woocommerce/woocommerce
# Migrate `@woocommerce/components/search` to TS

### All Submissions:






### Changes proposed in this Pull Request:



Closes #35373.

- Migrate Search component to TypeScript
- Add `@types/prop-types`

> You can pull up a diff by running git difftool <target branch>:<path of new file> <source branch>:<path of old file> - p1666772667777599-slack-C01SFMVEYAK







### How to test the changes in this Pull Request:



1. Run `pnpm --filter=@woocommerce/components build`
2. Run `pnpm --filter=woocommerce/client/admin ts:check`
3. Create some products and orders
4. Perform a smoke test on analytic pages, especially the **Advanced filters** area



### Other information:




### FOR PR REVIEWER ONLY:
